querydsl-mongodb/src/main/java/com/querydsl/mongodb/MongodbSerializer.java

```diff
@@ -127,6 +127,11 @@ public Object visit(Operation<?> expr, Void context) {
             if (subOp == Ops.IN) {
                 return visit(OperationImpl.create(Boolean.class, Ops.NOT_IN, subOperation.getArg(0),
                         subOperation.getArg(1)), context);
+            } else if (subOp == Ops.COL_IS_EMPTY) {
```

Comment:
Yes, that's what I mean, and only on the first level. Replacing the top level $or with $nor should work.